### PR TITLE
Php3branch > tl1revbranch

### PR DESCRIPTION
**Pull request by php-3**

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/company/apigetcompanydata.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+$response = array();
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+
+    if (__DEBUG__) {
+        echo '<br>if post request method</br>';
+    }
+    $objcompany = new clsCompany();
+    if (__DEBUG__) {
+        echo '<br>check object from clsCompany?? </br>';
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/company/apigetcompanydata.php]: debug message should be descriptive
"clsCompany object created"

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/company/clscompany.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public $db , $conf;
+
+    function __construct() {
+        global $db, $conf;
+        $this->db = $db;
+        $this->conf = $conf ;
+    }
+
+    public function fnGetCompanyData() {
+        echo $conf->global->MAIN_INFO_SOCIETE_NOM;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/company/clscompany.php]: why echo???

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/apigetproductsandservices.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+  Crreated by amera.ahmed dev.php3 date : 23/1/2018
+ */
+include_once '../../../../master.inc.php';
+include_once DOL_DOCUMENT_ROOT . '/custom/constants.php';
+include_once DOL_DOCUMENT_ROOT . '/custom/extintegration/app/products_services/clsproductandservices.php';
+$response = array();
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/apigetproductsandservices.php]: better at start of any API is to set a debug message "APINAME has been called"

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    public function fnGetProductsAndServices($type) {
+        try {
+            if (__DEBUG__) {
+                echo '</br>fnGetProducts_Services() begin ? </br>';
+            }
+            $sql = $this->fnGetTableQuery($type);
+            $result = $this->db->query($sql);
+            if ($result) {
+                $i = mysqli_num_rows($result);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php]: don't use
mysqli_num_rows

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    public function fnGetProductsAndServicesData($id) {
+        try {
+            if (__DEBUG__) {
+                echo '</br>fnGetProducts_Services() begin ? </br>';
+            }
+            $sql = $this->fnGetDataQuery($id);
+            $result = $this->db->query($sql);
+            if ($result) {
+                $i = mysqli_num_rows($result);
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php]: if u didn't used $I so why you define it?

Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php]: don't use mysqli_num_rows

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                if (mysqli_num_rows($result) == 0) {
+                    $obj = RS_IS_NULL;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php]: RS_IS_NULL should be only used for result sets not for objects

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/thirdparty/apigetthirdparty.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $type = $_POST['type'];
+
+    if (__DEBUG__) {
+        echo '<br>if post request method</br>';
+    }
+
+    if (isset($type)) {
+
+        $objthirdparty = new clsthirdparty();
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/thirdparty/apigetthirdparty.php]: clsthirdparty
clsThirdParty

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/thirdparty/apigetthirdpartydata.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $id = $_POST['id'];
+
+    if (__DEBUG__) {
+        echo '<br>if post request method</br>';
+    }
+
+    if (isset($id)) {
+
+        $objthirdparty = new clsthirdparty();
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/thirdparty/apigetthirdpartydata.php]: clsthirdparty
clsThirdParty

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/thirdparty/clsthirdparty.php [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+/**
+ * cls that get all thirdparty data 
+ * amera.ahmed dev.php3 date : 23/1/2018
+ */
+if (__DEBUG__) {
+    echo '</br>clsthirdparty has been included</br>';
+}
+
+class clsthirdparty {
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/thirdparty/clsthirdparty.php]: clsthirdparty
clsThirdParty

File: portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php (hunk 5 above)
Comment (php-3) [reply portfolio/ifintech/v1/ierp/htdocs/custom/extintegration/app/products_services/clsproductandservices.php]: I used $i in for loop

**Pull request merged**